lib/update/utils.go

```diff
@@ -96,3 +90,24 @@ func planetNeedsUpdate(profile string, installed, update appservice.Application)
 	updateNewer := updateVersion.Compare(*version) > 0
 	return updateNewer, nil
 }
+
+func getRuntimePackage(manifest schema.Manifest, profile schema.NodeProfile, clusterRole schema.ServiceRole) (*loc.Locator, error) {
+	runtimePackage, err := manifest.RuntimePackage(profile)
+	if err != nil && !trace.IsNotFound(err) {
+		return nil, trace.Wrap(err)
+	}
+	if err == nil {
+		return runtimePackage, nil
+	}
+	// Look for legacy package
+	packageName := loc.LegacyPlanetMaster.Name
+	if clusterRole == schema.ServiceRoleNode {
+		packageName = loc.LegacyPlanetNode.Name
+	}
+	runtimePackage, err = manifest.Dependencies.ByName(packageName)
+	if err != nil {
+		logrus.Warnf("Failed to find the legacy runtime package in manifest: %v.", err)
+		return nil, trace.NotFound("runtime package not found")
```

Comment:
I would make error message a bit more verbose, at least add profile/role information. Same for the log entry above.